src/indexes/text/proximity.h

```diff
@@ -62,6 +62,12 @@ class ProximityIterator : public TextIterator {
   const PositionRange& CurrentPosition() const override;
   bool NextPosition() override;
   FieldMaskPredicate CurrentFieldMask() const override;
+  // Returns true if iterator is at a valid state with current key, position,
+  // and field.
+  bool IsIteratorValid() const override {
```

Comment:
This API needs to be aware of require_positions. It is incorrect to check for current_position_ when require_positions is false